models/weight_recorder.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+ *  weight_recorder.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "weight_recorder.h"
+
+// C++ includes:
+#include <numeric>
+
+// Includes from libnestutil:
+#include "compose.hpp"
+#include "logging.h"
+
+// Includes from nestkernel:
+#include "event_delivery_manager_impl.h"
+#include "kernel_manager.h"
+#include "sibling_container.h"
+
+// Includes from sli:
+#include "arraydatum.h"
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+#include "integerdatum.h"
+
+nest::weight_recorder::weight_recorder()
+  : Node()
+  // record time, gid, weight and receiver gid
+  , device_( *this,
+      RecordingDevice::SPIKE_DETECTOR,
+      "csv",
```

Comment:
I would like to keep `csv` and in my opinion it is not misleading. It is very common to have a tab-separated csv file. Every csv reader I've seen so far (i.e. in Python, Excel, LibreOffice, etc) has the option to choose the delimiter. Often this is even done automatically.

The advantage of using a known file extension like 'csv' is that user immediately knows how to open the file.